x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/ConfigInitialNode.java

```diff
@@ -418,16 +435,22 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
                 e.addSuppressed(ex);
             }
             try {
-                Files.deleteIfExists(instantAutoConfigDir);
+                deleteDirectory(instantAutoConfigDir);
             } catch (Exception ex) {
                 e.addSuppressed(ex);
             }
             throw e;
```

Comment:
I was trying to force an exception in the try block from above, and the easiest way that I could think of was to mess with the with the classpath.
This made me realize that such exceptions are actually errors:
```
Exception in thread "main" java.lang.NoClassDefFoundError: org/bouncycastle/asn1/ASN1Encodable
        at org.elasticsearch.xpack.security.cli.ConfigInitialNode.execute(ConfigInitialNode.java:190)
        at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:75)
        at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:114)
        at org.elasticsearch.cli.Command.main(Command.java:79)
        at org.elasticsearch.xpack.security.cli.ConfigInitialNode.main(ConfigInitialNode.java:102)
Caused by: java.lang.ClassNotFoundException: org.bouncycastle.asn1.ASN1Encodable
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:182)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
        ... 5 more
```
and that we don't run the clean-up after errors.
But I think we should, so I suggest we replace `} catch(Exception e) {` with `} catch(Throwable t) {` in all places where we do  the cleanups.

Reply:
sounds good to me, I hadn't thought about that too until I saw the stacktrace above